eip_operator/src/main.rs

```diff
@@ -159,19 +165,23 @@ async fn run() -> Result<(), Error> {
 
     tasks.push({
         let eip_api = eip_api.clone();
+        let (eips, eip_notifications) = pin!(make_reflector::<Eip>(&eip_api)).await;
         let node_api = node_api.clone();
-        let watch_config = kube_runtime::watcher::Config::default();
 
         task::spawn(async move {
-            let mut watcher = pin!(kube_runtime::watcher(eip_api.clone(), watch_config));
-
-            while let Some(eip_event) = watcher.try_next().await.unwrap_or_else(|e| {
-                event!(Level::ERROR, err = %e, "Error watching eips");
-                None
-            }) {
-                if let kube_runtime::watcher::Event::Apply(eip)
-                | kube_runtime::watcher::Event::Delete(eip) = eip_event
-                {
+            // Repeating timer to label egress nodes
+            // This allows the egress node to be re-labeled even if a new EIP event is missed
+            let interval = time::interval(Duration::from_secs(300));
+            // timer needs to emit a `()` value for the select_all below
+            let repeating_timer = IntervalStream::new(interval).map(|_| ()).boxed();
```

Comment:
at the point we're writing our own time-based reconciler + reflector, it now feels like we've reinvented the kube-rs `Controller`, where we can return a requeue event for each Eip resource after it's reconciled to reschedule it for the future. I don't think we need our k8s-operator wrapper here that adds in finalizers, but I wonder if a plain Controller would work here?

Reply:
I wonder if we should make the finalizer an `Option`. This isn't the first time that I've wanted controller-style reconciliations, but without the finalizer added.